src/main/java/com/echobox/logging/escalator/EscalationLogger.java

```diff
@@ -0,0 +1,459 @@
+  public void trace(Marker marker, String format, Object arg1, Object arg2) {
+    checkForEscalation(LoggingLevel.TRACE, marker, format, arg1, arg2);
+    
+    logger.trace(marker, format, arg1, arg2);
+  }
+  
+  @Override
+  public void trace(Marker marker, String format, Object... argArray) {
+    checkForEscalation(LoggingLevel.TRACE, marker, format, argArray);
+    
+    logger.trace(marker, format, argArray);
+  }
+  
+  @Override
+  public void trace(Marker marker, String msg, Throwable throwable) {
+    checkForEscalation(LoggingLevel.TRACE, marker, msg, throwable);
+    
+    logger.trace(marker, msg, throwable);
+  }
+  
+  @Override
+  public boolean isDebugEnabled() {
+    return logger.isDebugEnabled();
+  }
+  
+  @Override
+  public boolean isDebugEnabled(Marker marker) {
+    return logger.isDebugEnabled(marker);
+  }
+  
+  @Override
+  public void debug(String msg) {
+    checkForEscalation(LoggingLevel.DEBUG, msg);
+    
+    logger.debug(msg);
+  }
+  
+  @Override
+  public void debug(String format, Object arg) {
+    checkForEscalation(LoggingLevel.DEBUG, format, arg);
+    
+    logger.debug(format, arg);
+  }
+  
+  @Override
+  public void debug(String format, Object arg1, Object arg2) {
+    checkForEscalation(LoggingLevel.DEBUG, format, arg1, arg2);
+    
+    logger.debug(format, arg1, arg2);
+  }
+  
+  @Override
+  public void debug(String format, Object... arguments) {
+    checkForEscalation(LoggingLevel.DEBUG, format, arguments);
+    
+    logger.debug(format, arguments);
+  }
+  
+  @Override
+  public void debug(String msg, Throwable throwable) {
+    checkForEscalation(LoggingLevel.DEBUG, msg, throwable);
+    
+    logger.debug(msg, throwable);
+  }
+  
+  @Override
+  public void debug(Marker marker, String msg) {
+    checkForEscalation(LoggingLevel.DEBUG, marker, msg);
+    
+    logger.debug(marker, msg);
+  }
+  
+  @Override
+  public void debug(Marker marker, String format, Object arg) {
+    checkForEscalation(LoggingLevel.DEBUG, marker, format, arg);
+    
+    logger.debug(marker, format, arg);
+  }
+  
+  @Override
+  public void debug(Marker marker, String format, Object arg1, Object arg2) {
+    checkForEscalation(LoggingLevel.DEBUG, marker, arg1, arg2);
+    
+    logger.debug(marker, format, arg1, arg2);
+  }
+  
+  @Override
+  public void debug(Marker marker, String format, Object... arguments) {
+    checkForEscalation(LoggingLevel.DEBUG, marker, format, arguments);
+    
+    logger.debug(marker, format, arguments);
+  }
+  
+  @Override
+  public void debug(Marker marker, String msg, Throwable throwable) {
+    checkForEscalation(LoggingLevel.DEBUG, marker, msg, throwable);
+    
+    logger.debug(marker, msg, throwable);
+  }
+  
+  @Override
+  public boolean isInfoEnabled() {
+    return logger.isInfoEnabled();
+  }
+  
+  @Override
+  public boolean isInfoEnabled(Marker marker) {
+    checkForEscalation(LoggingLevel.INFO, marker);
+    
+    return logger.isInfoEnabled(marker);
+  }
```

Comment:
Out of interest, why do we check for escalation in these methods?

Reply:
oh sorry that is a copy/paste issue - got a bit carried away!